src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ImplementStandardExceptionConstructors.cs

```diff
@@ -27,23 +31,96 @@ public abstract class ImplementStandardExceptionConstructorsAnalyzer : Diagnosti
+        protected abstract string GetParameterLessConstructorMessage(ISymbol symbol);
+        protected abstract string GetConstructorWithStringTypeParameter(ISymbol symbol);
+        protected abstract string GetConstructorWithStringAndExceptionTypeParameter(ISymbol symbol);
+
+        private void AnalyzeSymbol(ISymbol symbol, SymbolAnalysisContext context)
+        {
+            var namedTypeSymbol = context.Symbol as INamedTypeSymbol;
+
+            //Check if type derives from Exception type
+            if (namedTypeSymbol.BaseType.MetadataName == context.Compilation.GetTypeByMetadataName("System.Exception").MetadataName)
+            {
+                //Get the list of constructors
+                var constructors = namedTypeSymbol.Constructors;
+
+                //Set flags for the 3 different constructos, that is being searched for
+                var defaultConstructorFound = false; //flag for default constructor
+                var secondConstructorFound = false; //flag for constructor with string type parameter
+                var thirdConstructorFound = false; //flag for constructor with string and exception type parameter
+
+                foreach (var ctor in constructors)
+                {
+                    var parameters = ctor.GetParameters();
+
+                    //case 1: Default constructor - no parameters
+                    if (parameters.Length == 0)
+                    {
+                        defaultConstructorFound = true;
+                    }
+                    //case 2: Constructor with string type parameter
+                    else if (parameters.Length == 1 && parameters[0].Type.SpecialType == SpecialType.System_String)
+                    {
+                        secondConstructorFound = true;
+                    }
+                    //case 3: Constructor with string type and exception type parameter
+                    else if (parameters.Length == 2 && parameters[0].Type.SpecialType == SpecialType.System_String &&
+                            parameters[1].Type == context.Compilation.GetTypeByMetadataName("System.Exception"))
+                    {
+                        thirdConstructorFound = true;
+                    }
+
+                    if (defaultConstructorFound && secondConstructorFound && thirdConstructorFound)
+                    {
+                        //reaches here only when all 3 constructors are found - no diagnostic needed 
+                        return;
+                    }
+
+                } //end of for loop
+
+                if (!defaultConstructorFound)
+                {
+                    //Missing default constructor - report diagnostic
+                    Diagnostic diagnostic;
+                    diagnostic = Diagnostic.Create(MissingConstructorRule, symbol.Locations[0], symbol.Name, GetParameterLessConstructorMessage(symbol));
```

Comment:
You can write `symbol.CreateDiagnostic(MissingConstructorRule, symbol.Name, ...)` That will fire the diagnostic in the right location.